streamer/src/bidirectional_channel.rs

```diff
@@ -0,0 +1,396 @@
+// this is a service to handle bidirections quinn channel
+use bincode::serialize;
+use crossbeam_channel::{Receiver, Sender};
+use quinn::SendStream;
+use serde_derive::{Deserialize, Serialize};
+use solana_perf::packet::{Packet, PacketBatch};
+use solana_sdk::signature::Signature;
+use std::{
+    collections::HashMap,
+    net::SocketAddr,
+    sync::{Arc, RwLock},
+};
+
+#[derive(Serialize, Deserialize)]
+pub struct QuicReplyMessage {
+    pub transaction_signature: Signature,
+    pub message: String,
+}
+
+pub struct QuicBidirectionalData {
+    pub transaction_signature_map: HashMap<Signature, u64>,
+    pub sender_ids_map: HashMap<u64, Sender<QuicReplyMessage>>,
+    pub sender_socket_address_map: HashMap<SocketAddr, u64>,
+    pub last_id: u64,
+}
+
+#[derive(Clone)]
+pub struct QuicBidirectionalReplyService {
+    data: Arc<RwLock<QuicBidirectionalData>>,
+    pub service_reciever: Receiver<QuicReplyMessage>,
+    pub service_sender: Sender<QuicReplyMessage>,
+}
+
+pub fn get_signature_from_packet(packet: &Packet) -> Option<Signature> {
+    // add instruction signature for message
+    match packet.data(1..65) {
+        Some(signature_bytes) => {
+            let sig = Signature::new(&signature_bytes);
+            Some(sig)
+        }
+        None => None,
+    }
+}
+
+impl QuicBidirectionalReplyService {
+    pub fn new() -> Self {
+        let (service_sender, service_reciever) = crossbeam_channel::unbounded();
+        Self {
+            service_reciever: service_reciever,
+            service_sender: service_sender,
+            data: Arc::new(RwLock::new(QuicBidirectionalData {
+                transaction_signature_map: HashMap::new(),
+                sender_ids_map: HashMap::new(),
+                sender_socket_address_map: HashMap::new(),
+                last_id: 1,
+            })),
+        }
+    }
+
+    pub fn send_message(&self, transaction_signature: &Signature, message: String) {
+        let _ = self.service_sender.send(QuicReplyMessage {
+            transaction_signature: *transaction_signature,
+            message,
+        });
+    }
+
+    pub fn add_stream(&self, quic_address: &SocketAddr, send_stream: SendStream) {
+        let (sender_channel, reciever_channel) = crossbeam_channel::bounded(256);
```

Comment:
needs comment what happens when channel bound is reached and why you expect this to be below 256

Reply:
made it unbounded for now.